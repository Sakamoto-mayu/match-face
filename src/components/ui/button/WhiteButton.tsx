'use client';
import { ComponentProps } from 'react';
import { twMerge } from 'tailwind-merge';

const WhiteButton = ({ label, error, ...props }: W_BtnProps) => {
  // error=trueの場合、font & border を赤に
  const baseStyle = error
    ? 'bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md border-red-600 text-red-600 '
    : 'bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md ';
  const mergeStyle = twMerge(baseStyle + props.className);
  return (
    <button {...props} className={mergeStyle}>
      {label}
    </button>
  );
};

type W_BtnProps = {
  label: string;
  error?: boolean;
} & ComponentProps<'button'>;

export default WhiteButton;
