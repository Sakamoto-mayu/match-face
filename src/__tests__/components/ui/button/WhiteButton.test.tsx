import WhiteButton from "@/components/ui/button/WhiteButton";
import { cleanup, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("WhiteButton.tsx", () => {
  beforeAll(() => {
    jest.clearAllMocks();
    cleanup();
  });
  afterAll(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it("スナップショットテスト", () => {
    const { container } = render(<WhiteButton label="テスト" />);
    expect(container).toMatchSnapshot();
    cleanup();
  });
  describe("propsテスト", () => {
    afterEach(() => {
      cleanup();
    });
    it("props.labelの通りにラベルがつく", () => {
      render(<WhiteButton label="testing" />);
      const LabeledBtn = screen.getByRole("button", { name: "testing" });
      expect(LabeledBtn).toBeInTheDocument();
    });
    it("props.error = true の場合、borderと文字が赤くなる", () => {
      // クラスに"border-red-600" "text-red-600"があるかテスト
      render(<WhiteButton label="テスト" error={true} />);
      const ErrorBtn = screen.getByRole("button");
      expect(ErrorBtn).toHaveClass("border-red-600");
      expect(ErrorBtn).toHaveClass("text-red-600");
    });
    it("twMergeが正常に動作している", () => {
        // Reactコンポーネントのデフォルトのpropsが渡せるかもついでにテスト
        render(<WhiteButton label="テスト" className="test-class" />);
        const ErrorBtn = screen.getByRole("button");
        expect(ErrorBtn).toHaveClass("test-class");
      });
  });
});
