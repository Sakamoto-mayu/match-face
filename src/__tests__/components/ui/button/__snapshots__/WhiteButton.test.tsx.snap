// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhiteButton.tsx スナップショットテスト 1`] = `
<div>
  <button
    class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md"
  >
    テスト
  </button>
</div>
`;
